docs/proposals/Disclosure/README.md

```diff
@@ -0,0 +1,111 @@
+# Disclosure
+
+A `<Disclosure>` component will reveal and hide additional information that our
+users may care about, should they choose to learn more about it.
+
+## Design Usage and Guidelines
+
+- use when you want to reduce feelings of being overwhelmed and to reduce
+  distractions
+- used for lightweight editing, such as selecting a value for a setting
+- used for information that is lower priority or don’t need to see all the time
+- for progressive disclosure of elements that aren't essential for the user to
+  view all at once
+- not be used for other critical information that requires immediate action
+
+  ### How it might behave:
+
+  - both title bar and arrow are clickable
+    - an interactive title bar that has relevant info about its hidden content
+    - arrow indicating the next direction if clicked
+  - panel is closed on default
+  - a prop that controls the revealing / hiding state
+    - when “showing”, it will roll down and reveal the inner contents
+    - when “hiding”, it will roll back up to hide the contents
+      - at hidden state, content should have no height and not
+        clickable/interactable
+      - title has a `<summary>` of its content
+
+## Content Guidelines
+
+- **Title** should be informative and label the type of content grouped in the
+  body content in it
+  - use `<Heading level={4}>` component
+- **Content** should be actionable and clear. May contain:
+  - good ol’ plain text
+  - accepts any React component as a child
+  - except: `<Page>`, `<Table>`
+
+## Accessibility
+
+- Users should be able to use their keyboard and toggle the component's
+  open/close state
+- Headings are permitted in `<summary>` elements to provide in-page navigational
+  assistance
+- Include `aria-expanded` attribute on the trigger to communicate to assistive
+  technology
+- Include `aria-controls` attribute ties on the trigger to the content it
+  controls using the `id` of the collapsible container
+- The trigger contains a downward-pointing-arrow to hint that it can be
+  expanded. when the disclosure item is in expanded state, this is rotated 180
+  degrees to point upwards
+- The icon will be given an `aria-hidden=”true”` attribute to hide it from
+  assistive technologies, as well as `focusable=”false”` to address an
+  inconsistency in IE and older versions of Edge
+
+## Responsiveness
+
+- This component should handle both click and touch, as well as keyboard inputs
+- This component should fill the width of its container
+- If the component is less than `375px` wide, the title will wrap and should not
+  be truncated
+- Text increases/decreases when user zooms in/out
+
+## Mockup
+
+!["Disclosure Mockup"](https://gist.githubusercontent.com/kingstonfung/b882aa211f4213b1ac89bd515dc48a50/raw/efef596279fa0070d26b67cd364e533ca3d572bd/image2.png)
+
+## Interface
+
+```jsx
+<Disclosure title="Advanced Options">
+  <Content>Example content here</Content>
```

Comment:
One thing we could clarify here or at the actual code review level is whether we want to have a [`<Content>` component](https://atlantis.getjobber.com/components/content) in here to provide spacing of the elements by default, or if <Content> in this example just means "children"

Reply:
I think we can just do `Example content here` without the `<Content>` for clarity. We intended to meant just "children" and not necessary have to be `<Content>` wrapped.